### PR TITLE
BZ#1891515 - Clarify OVN-Kubernetes egress IP address platform support

### PR DESCRIPTION
**Pull request by jboxman**
- https://bugzilla.redhat.com/show_bug.cgi?id=1891515

@alexanderConstantinescu, the update will live here once identified.

@vikram-redhat, this update will need to go in to clarify platform support for this feature in 4.6.

Preview:

- [Configuring an egress IP address](https://bz-1891515--ocpdocs.netlify.app/openshift-enterprise/latest/networking/ovn_kubernetes_network_provider/configuring-egress-ips-ovn.html)

### CODE REVIEW

**Review by vikram-redhat — APPROVED** [review]

File: modules/nw-egress-ips-about.adoc [hunk 1]
```diff
@@ -26,6 +26,15 @@ An egress IP address is implemented as an additional IP address on the primary n
 [id="nw-egress-ips-platform-support_{context}"]
 == Platform support
 
+// Clarify platform support -
+// https://bugzilla.redhat.com/show_bug.cgi?id=1891515
+
+////
+[IMPORTANT]
+====
+Currently, the egress IP addresses are not compatible with Amazon AWS, Azure Cloud, or any other cloud platform that does not support layer 2 manipulations due to their incompatibility with macvlan traffic.
```
Comment (alexanderConstantinescu) [inline modules/nw-egress-ips-about.adoc:35]: I am not sure on the exact wording for this, but we should get rid of `or any other cloud platform that does not support layer 2 manipulations due to their incompatibility with macvlan traffic.`. I am thinking something along the lines of:

```
Currently, the fully integrated egress IP solution of OpenShift is only supported on bare-metal and vSphere. OpenShift's egress IP is not expected to work on or tested on: Amazon AWS, Azure Cloud, and Google GCP which all require additional system administrator setup using the individual cloud console. 
```

**Comment by jboxman** [comment]
@alexanderConstantinescu, I've updated the language; maybe this is more suitable.

File: modules/nw-egress-ips-about.adoc [hunk 2]
```diff
@@ -26,16 +26,28 @@ An egress IP address is implemented as an additional IP address on the primary n
...
 == Platform support
 
-While you can use the egress IP address functionality on any platform that {product-title} supports, the following platform specific considerations apply:
+Support for the egress IP address functionality on various platforms is summarized in the following table:
 
-Cloud platforms::
-Additional IP addresses must be assigned manually to appropriate nodes by an administrator.
+[IMPORTANT]
+====
+The egress IP address implementation is not compatible with Amazon AWS, Azure Cloud, or any other public cloud platform that does not support manipulation of layer 2 network traffic.
```
Comment (alexanderConstantinescu) [inline modules/nw-egress-ips-about.adoc:33]: I would rephrase to `or any other public cloud platform incompatible with the automatic layer 2 network manipulation required by OpenShift's egress IP feature.`. The reason is that in the future this will work, it's just that we currently have not done the necessary development for that to happen.

File: modules/nw-egress-ips-about.adoc [hunk 3]
```diff
@@ -26,16 +26,28 @@ An egress IP address is implemented as an additional IP address on the primary n
...
+| vSphere | Yes
+| {rhosp} | No
+| Public cloud^[1]^ | No
 
-vSphere::
-Additional configuration by an administrator is necessary to allow {product-title} to automatically assign additional IP addresses to nodes.
+|===
+[small]
+--
+1. Public cloud platforms, such as Amazon AWS and Google GCP, do not support automatic assignment of an egress IP address.
```
Comment (alexanderConstantinescu) [inline modules/nw-egress-ips-about.adoc:49]: I would rephrase here too, to the following: `do not support automatic assignment of an egress IP address following OpenShift's current assignment procedure.`. For the same reason I mentioned above.

File: modules/nw-egress-ips-about.adoc [hunk 4]
```diff
@@ -26,16 +26,28 @@ An egress IP address is implemented as an additional IP address on the primary n
...
+====
+
+[cols="1,1",options="header"]
+|===
+
+| Platform | Supported
 
-{rhosp}::
-Additional configuration by an administrator is necessary to allow {product-title} to automatically assign additional IP addresses to nodes.
+| bare metal | Yes
```
Comment (bergerhoffer) [inline modules/nw-egress-ips-about.adoc:41]: bare metal -> Bare metal

File: modules/nw-egress-ips-about.adoc [hunk 5]
```diff
@@ -26,16 +26,28 @@ An egress IP address is implemented as an additional IP address on the primary n
...
+| vSphere | Yes
+| {rhosp} | No
+| Public cloud^[1]^ | No
 
-vSphere::
-Additional configuration by an administrator is necessary to allow {product-title} to automatically assign additional IP addresses to nodes.
+|===
+[small]
+--
+1. Public cloud platforms, such as AWS and Google Cloud Platform (GCP), do not support the automatic layer 2 network manipulation required by the egress IP feature.
```
Comment (bergerhoffer) [inline modules/nw-egress-ips-about.adoc:49]: Not sure if it matters, but you use different examples here than in the NOTE above the table. Also, is it necessary to have that same warning both places?

Comment (bergerhoffer) [reply modules/nw-egress-ips-about.adoc:49]: However looking again, you already have "No" in the column for whether it's supported, so why would this note be necessary saying that these cloud platforms don't support the feature? The NOTE above made it sound like SOME public clouds might be supported if they have the "automatic layer 2 network manipulation required by the egress IP feature."

**Review by bergerhoffer — COMMENTED** [review]
Just some thoughts

File: modules/nw-egress-ips-about.adoc (hunk 5 above)
Comment (jboxman) [reply modules/nw-egress-ips-about.adoc:49]: @bergerhoffer, so while this feature might work with manual intervention for IP address assignment on some cloud providers, officially it was never tested by QE, so we don't actually support it on any cloud platform. And even if it did otherwise work, without automatic address assignment, it's incomplete.

There's probably no reason to include the footnote as well, though. Mostly it just looks awesome.
